language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -109,6 +109,78 @@ object CodeGenRunner extends StrictLogging {
     logger.warn(s"Finish writing file '$outputFile'")
   }
 
+  /** Given the package prefixes specified per DAR and the module-prefixes specified in
+    * daml.yaml, produce the combined prefixes per package id.
+    */
+  def resolvePackagePrefixes(
+      pkgPrefixes: Map[PackageId, String],
+      modulePrefixes: Map[PackageReference, String],
+      interfaces: Seq[Interface],
+  ): Map[PackageId, String] = {
+    val metadata: Map[PackageReference.NameVersion, PackageId] = interfaces.view
+      .flatMap(iface =>
+        iface.metadata.iterator.map(metadata =>
+          PackageReference.NameVersion(metadata.name, metadata.version) -> iface.packageId
+        )
+      )
+      .toMap
+    def resolveRef(ref: PackageReference): PackageId = ref match {
+      case nameVersion: PackageReference.NameVersion =>
+        metadata.getOrElse(
+          nameVersion,
+          throw new IllegalArgumentException(
+            s"""No package $nameVersion found, available packages: ${metadata.keys.mkString(
+              ", "
+            )}"""
+          ),
+        )
+    }
+    val resolvedModulePrefixes: Map[PackageId, String] = modulePrefixes.view.map { case (k, v) =>
+      resolveRef(k) -> v.toLowerCase
+    }.toMap
```

Comment:
```suggestion
    val resolvedModulePrefixes: Map[PackageId, String] = modulePrefixes.map { case (k, v) =>
      resolveRef(k) -> v.toLowerCase
    }
```
No need to go through a view for a single transformation where the input and output types are the same, right?

Reply:
My kingdom for a compiler with proper fusion so I don’t have to worry about this 497eacbc0ff1bda4b15401d3ad7856185b551eae

Reply:
There is one but I heard you complaining about segfaults there. :trollface: